frontend/src/core/cells/utils.ts

```diff
@@ -37,8 +37,8 @@ export function notebookNeedsSave(
   const names = data.map((d) => d.name);
   return (
     !arrayShallowEquals(codes, lastSavedNotebook.codes) ||
-    !arrayShallowEquals(configs, lastSavedNotebook.configs) ||
     !arrayShallowEquals(names, lastSavedNotebook.names) ||
+    !isEqual(configs, lastSavedNotebook.configs) ||
```

Comment:
Not sure why this was working with `arrayShallowEquals` before ...

Reply:
possibly mutating state instead of keeping it immutable